### PR TITLE
8.0 Add sale_rental_lot_selection + several fixes and improvements in sale_rental

### PR DESCRIPTION
**Pull request by alexis-via**

### CODE REVIEW

**Comment by alexis-via** [comment]
@eLBati I made the work required to have travis green with my module sale_rental_lot_selection. At last !

**Comment by eLBati** [comment]
@alexis-via hi, github says "This branch has conflicts that must be resolved". Could you check?

**Comment by alexis-via** [comment]
mmm, the conflict is in product_id_change_with_wh... so it has to be studied in depth and I need to re-test everything :-(

File: sale_order_lot_selection/model/sale.py [hunk 1]
```diff
@@ -45,24 +55,28 @@ def product_id_change_with_wh(
...
                 })._product_available()
-            if qty_res[product]['qty_available'] > 0:
+            if qty_res[product.id]['qty_available'] > 0:
                 if lot.id not in available_lots:
                     available_lots.append(lot.id)
-        res.update({'domain': {'lot_id': [('id', 'in', available_lots)]}})
+        res.update({'domain': {'lot_id': [
+            ('id', 'in', available_lots),
+            ('product_id', '=', product.id),
+            ]}})
```
Comment (jabibi) [inline sale_order_lot_selection/model/sale.py]: I tested with this piece of code and seems to work :ok_hand: 
Please look at the setdefault part in order to maintain previous values

```
    @api.multi
    def product_id_change_with_wh(
            self, pricelist, product_id, qty=0,
            uom=False, qty_uos=0, uos=False, name='', partner_id=False,
            lang=False, update_tax=True, date_order=False, packaging=False,
            fiscal_position=False, flag=False, warehouse_id=False):
        res = super(SaleOrderLine, self).product_id_change_with_wh(
            pricelist, product_id, qty=qty, uom=uom,
            qty_uos=qty_uos, uos=uos, name=name, partner_id=partner_id,
            lang=lang, update_tax=update_tax,
            date_order=date_order, packaging=packaging,
            fiscal_position=fiscal_position, flag=flag,
            warehouse_id=warehouse_id)

        product = self.env['product.product'].browse(product_id)
        warehouse = self.env['stock.warehouse'].browse(warehouse_id)
        location = self._select_lot_stock_location(product, warehouse)
        # DO NOT use 'product_id' AFTER THIS LIMIT ; only use 'product'
        product = self._select_lot_product(product)

        # Search all lot existing lot for the product and location selected
        quants = self.env['stock.quant'].read_group([
            ('product_id', '=', product.id),
            ('location_id', 'child_of', location.id),
            ('qty', '>', 0),
            ('lot_id', '!=', False),
            ], ['lot_id'], 'lot_id')
        available_lots = [quant['lot_id'][0] for quant in quants]
        res.setdefault('domain', {}).update(
            {'lot_id': [('id', 'in', available_lots)]})
        res['value']['lot_id'] = False
        return res
```

**Comment by jabibi** [comment]
I have some minor fixes / improvements that if you want I can push to your branch

**Comment by alexis-via** [comment]
@jabibi Yes, don't hesitate to make a PR to my branch.

**Comment by jabibi** [comment]
PR akretion/sale-workflow#4

I did not say, but you do a great job @alexis-via

**Comment by pedrobaeza** [comment]
Please fix conflicts

**Comment by rousseldenis** [comment]
Closing this as no activity. @alexis-via Please feel free to reopen it if you still need it. Take care of your PR's

**Pull request closed without merging**